plugins/languages/otter.nix

```
@@ -0,0 +1,58 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "otter";
+  originalName = "otter.nvim";
+  defaultPackage = pkgs.vimPlugins.otter-nvim;
+
+  maintainers = [ lib.maintainers.perchun ];
+
+  settingsOptions = {
+    lsp = {
+      hover = {
+        border = helpers.defaultNullOpts.mkListOf lib.types.str [
+          "╭"
+          "─"
+          "╮"
+          "│"
+          "╯"
+          "─"
+          "╰"
+          "│"
+        ] "";
+      };
+      diagnostic_update_events = helpers.defaultNullOpts.mkListOf lib.types.str [ "BufWritePost" ] ''
+        `:h events` that cause the diagnostics to update. Set to:
+        [ "BufWritePost" "InsertLeave" "TextChanged" ] for less performant
+        but more instant diagnostic updates
+      '';
+    };
+    buffers = {
+      set_filetype = helpers.defaultNullOpts.mkBool false ''
+        if set to true, the filetype of the otterbuffers will be set.
+        otherwise only the autocommand of lspconfig that attaches
+        the language server will be executed without setting the filetype
+      '';
+      write_to_disk = helpers.defaultNullOpts.mkBool false ''
+        write <path>.otter.<embedded language extension> files
+        to disk on save of main buffer.
+        useful for some linters that require actual files
+        otter files are deleted on quit or main buffer close
+      '';
+    };
+    strip_wrapping_quote_characters = helpers.defaultNullOpts.mkListOf lib.types.str [
```

```suggestion
    };

    strip_wrapping_quote_characters = helpers.defaultNullOpts.mkListOf lib.types.str [
```

Generally, we prefer a blank line between each option, unless it's (e.g.) a small submodule with so few options that it's more readable without.

Not a big deal, I wouldn't have mentioned it if I wasn't already asking for changes :smile: